acceptance/lib/pxp-agent/test_helper.rb

```diff
@@ -265,16 +289,20 @@ def rpc_blocking_request(broker, targets,
   end
 
   message = PCP::Message.new({
-    :message_type => 'http://puppetlabs.com/rpc_blocking_request',
+    :message_type => blocking ? 'http://puppetlabs.com/rpc_blocking_request' : 'http://puppetlabs.com/rpc_non_blocking_request',
     :targets => targets
   })
 
-  message.data = {
+  message_data = {
```

Comment:
Yes these helper methods assume that the messages hitting the handler are the response for the expected transaction. This is reasonably safe as the client identity only exists for the duration of this method; so it shouldn't receive messages for anything other than this RPC action.

I will raise a separate ticket for the fact that we do not have full traceability of the transaction_id; it's code that existed before this ticket and shouldn't be addressed in this PR.